packages/insured-bridge-relayer/src/index.ts

```diff
@@ -75,11 +75,20 @@ export async function run(logger: winston.Logger, l1Web3: Web3): Promise<void> {
     // method will error if the bot runner has specified a L1 tokens that is not part of the Bridge Admin whitelist.
     await approveL1Tokens(logger, l1Web3, gasEstimator, accounts[0], config.bridgeAdmin, filteredL1Whitelist);
 
+    // Construct the profitability calculator based on the filteredL1Whitelist and relayerDiscount.
+    const profitabilityCalculator = new ProfitabilityCalculator(
+      logger,
+      filteredL1Whitelist,
+      await l1Web3.eth.getChainId(),
+      config.relayerDiscount
+    );
+
     const relayer = new Relayer(
       logger,
       gasEstimator,
       l1Client,
       l2Client,
+      profitabilityCalculator,
```

Comment:
Doesn't `profitabilityCalculator.update()` need to be added to the first `await Promise.all([])` in the main infinite loop?

Reply:
yup! good catch

Reply:
I still don't see this in `update` have you pushed up all changes?

Reply:
it's there sir. see line 118.